src/deepsparse/v2/text_generation/nl_engine_operator.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import os
+from typing import Any, List, Optional, Tuple
+
+from pydantic import BaseModel, Field
+
+from deepsparse.utils.onnx import (
+    CACHE_INPUT_PREFIX,
+    overwrite_onnx_model_inputs_for_kv_cache_models,
+)
+from deepsparse.v2.operators.engine_operator import DEEPSPARSE_ENGINE, EngineOperator
+from deepsparse.v2.utils import Context, InferenceState, PipelineState
+
+
+__all__ = ["NLEngineOperator"]
+
+
+class NlEngineInput(BaseModel):
+    engine_inputs: List = Field(description="engine inputs")
+    kv_cache: Any = Field(description="kv_cache object")
+    tokens: List = Field(description="tokens")
+
+
+class NLEngineOperator(EngineOperator):
```

Comment:
this looks great - super happy with how this turned out and how we're splitting ownership of the various pipeline pieces